src/orchestration/Orchestration.ts

```diff
@@ -415,16 +415,18 @@ export class Orchestration {
             this.config
         );
 
-        if (this.config.identityPoolId && this.config.guestRoleArn) {
-            dispatch.setAwsCredentials(
-                new Authentication(this.config)
-                    .ChainAnonymousCredentialsProvider
-            );
-        } else if (this.config.identityPoolId) {
-            dispatch.setAwsCredentials(
-                new EnhancedAuthentication(this.config)
-                    .ChainAnonymousCredentialsProvider
-            );
+        if (this.eventCache.getSession(false)?.record) {
```

Comment:
suggestion: Please add comments that explain this portion

Reply:
Added comments!